src/gdb/gdb.c

```diff
@@ -144,32 +145,74 @@ void gdb_handle_exception(void)
 void parse_request(void)
 {
 	unsigned char *request;
+	unsigned int i;
 	int addr;
-
-	while (1) {
-		request = get_packet();
-		/* Log any exception caused by debug exception */
-		gdb_debug_info(request);
-
-		/* Pick incoming request handler */
-		unsigned char command = *request++;
-
-		switch (command) {
-		/* Continue normal program execution and leave debug handler */
-		case 'c':
-			if (hex_to_int(&request, &addr))
-				sregs[DEBUG_PC] = addr;
-
-			/* return from exception */
-			return;
+	int length;
+
+while (1) {
+	request = get_packet();
+	/* Log any exception caused by debug exception */
+	gdb_debug_info(request);
+	/* Pick incoming request handler */
+	unsigned char command = *request++;
+
+	switch (command) {
+	/* Continue normal program execution and leave debug handler */
+	case 'c':
+		if (hex_to_int(&request, &addr))
+			sregs[DEBUG_PC] = addr;
+
+		/* return from exception */
+		return;
+	/* insert breakpoint */
+	case 'Z':
+		switch (*request++) {
+		/* HW breakpoint */
+		case '1':
+			if (*request++ == ',' && hex_to_int(&request, &addr) &&
```

Comment:
can you comment here more what this block is looking for in request.

Reply:
Okay, when we receive packet/request from GDB it has specific format, for breakpoints it looks like "$ZX,ADDRESS,LEN,#CS" where:
Z - add breakpoint,
X - defines if this request is for software or hardware breakpoint. It takes values of zero or one and means software and hardware breakpoint respectively.
ADDRESS - at what address we which to put our breakpoint
LEN - address length
CS - checksum

So in above lines of code we go through this request and extract needed informations. 
Is it clear now?